GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -211,21 +216,28 @@ string ComposeOutput()
         /// <param name="executable">The executable from which to launch a process.</param>
         /// <param name="arguments">The arguments to pass to the executable.</param>
         /// <param name="input">Bytes to be written to the process's standard input stream, or <c>null</c> if no input is required.</param>
+        /// <param name="error">Bytes to be read from the process's standard error stream, or <c>null</c> if no output is required.</param>
         /// <param name="createWindow">A flag indicating whether a console window should be created and bound to the process.</param>
         /// <returns>A task that yields <c>true</c> if the process's exit code was zero, otherwise <c>false</c>.</returns>
         public static async Task<bool> RunCommandAsync(
             this IExecutable executable,
             ArgumentString arguments = default,
             byte[]? input = null,
+            byte[]? error = null,
             bool createWindow = false)
         {
-            using var process = executable.Start(arguments, createWindow: createWindow, redirectInput: input is not null);
+            using var process = executable.Start(arguments, createWindow: createWindow, redirectInput: input is not null, redirectOutput: error is not null, outputEncoding: error is not null ? Encoding.UTF8 : null);
             if (input is not null)
             {
                 await process.StandardInput.BaseStream.WriteAsync(input, 0, input.Length);
                 process.StandardInput.Close();
             }
 
+            if (error is not null)
+            {
+                await process.StandardError.BaseStream.ReadAsync(error, 0, error.Length);
+            }
```

Comment:
I can't ``process.StandardError.Close();`` either tests crashes.

Reply:
Sorry for the delay.
The output should be read *after* the process has finished / terminated, i.e. in `RunCommandAsync` after `await process.WaitForExitAsync()`.
As suggested in #6527, an argument `bool throwOnErrorOutput = true` should be added to `RunCommandAsync`. If `true`, the error output of the command shall be read and if not empty, be wrapped into a new `CommandErrorException` derived from `ExternalOperationException`. This exception does only need to be handled up the callstack unless the remaining output can be displayed despite the exception.

Reply:
I fixed the order to read the error output stream.
I suggest to leave throwOnErrorOutput to #6527 to reduce size and age of commit.
At the current time, my commit is based on #8802 and the original branch can't be merged anymore (BugReporter has been heavily refactored). I will try to rebase my commit based on master and not on #8802 if it's possible.